pkg/compose/convergence.go

```diff
@@ -328,7 +330,33 @@ func mustRecreate(expected types.ServiceConfig, actual moby.Container, policy st
 	}
 	configChanged := actual.Labels[api.ConfigHashLabel] != configHash
 	imageUpdated := actual.Labels[api.ImageDigestLabel] != expected.CustomLabels[api.ImageDigestLabel]
-	return configChanged || imageUpdated, nil
+	if configChanged || imageUpdated {
+		return true, nil
+	}
+
+	if c.networks != nil {
```

Comment:
Should we recreate if networks is nil but expected  has networks?

Reply:
networks is nil only if caller didn't created convergence with actual network state - this is the case running `compose run` command, need to investigate why we don't manage networks in this command, will do in a follow-up PR.